File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] indexes = new int[2];
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            if (map.containsKey(nums[i])) {
                indexes[0] = map.get(nums[i]);
                indexes[1] = i;
                return indexes;
            }
            map.put(target - nums[i], i);
        }
        return indexes;
    }
}

File: 0001-two-sum/0001-two-sum.kt
class Solution {
    fun twoSum(numbers: IntArray, target: Int): IntArray {
        val indexMap: MutableMap<Int, Int> = HashMap()
        for (i in numbers.indices) {
            val requiredNum = target - numbers[i]
            if (indexMap.containsKey(requiredNum)) {
                return intArrayOf(indexMap[requiredNum]!!, i)
            }
            indexMap[numbers[i]] = i
        }
        return intArrayOf(-1, -1)
    }
}
